src/Bundle/Maker/Factory/DoctrineObjectDefaultPropertiesGuesser.php

```diff
@@ -56,66 +57,125 @@ public function __invoke(MakeFactoryData $makeFactoryData, bool $allFields): voi
         /** @var ORMClassMetadata|ODMClassMetadata $metadata */
         $metadata = $em->getClassMetadata($class);
 
-        $dbType = $em instanceof EntityManagerInterface ? 'ORM' : 'ODM';
-
-        $this->guessDefaultValueForAssociativeFields($makeFactoryData, $metadata, $dbType, $allFields);
-        $this->guessDefaultValueForScalarFields($makeFactoryData, $metadata, $dbType, $allFields);
+        $metadata instanceof ORMClassMetadata && $this->guessDefaultValueForORMAssociativeFields($makeFactoryData, $metadata, $allFields);
+        $metadata instanceof ODMClassMetadata && $this->guessDefaultValueForODMAssociativeFields($makeFactoryData, $metadata, $allFields);
+        $metadata instanceof ORMClassMetadata && $this->guessDefaultValueForEmbedded($makeFactoryData, $metadata, $allFields);
+        $this->guessDefaultValueForScalarFields($makeFactoryData, $metadata, $allFields);
```

Comment:
maybe another clean strategy would have been to introduce `{ODM,ORM}DefaultGuesser`, WDYT? :thinking: